nifi-commons/nifi-utils/src/main/java/org/apache/nifi/stream/io/util/TextLineDemarcator.java

```diff
@@ -0,0 +1,227 @@
+                    if (crlfLength > 0) {
+                        i += crlfLength;
+                        if (crlfLength == 2) {
+                            lineLength++;
+                        }
+                        offsetInfo = new long[] { this.offset, lineLength, crlfLength, 1 };
+                        if (startsWith != null) {
+                            token = this.extractDataToken(lineLength);
+                        }
+                        this.index = i;
+                        this.mark = this.index;
+                        break lineLoop;
+                    }
+                }
+                this.index = i;
+            }
+        }
+        // EOF where last char(s) are not CRLF.
+        if (lineLength > 0 && offsetInfo == null) {
+            offsetInfo = new long[] { this.offset, lineLength, 0, 1 };
+            if (startsWith != null) {
+                token = this.extractDataToken(lineLength);
+            }
+        }
+        this.offset += lineLength;
+
+        // checks if the new line starts with 'startsWith' chars
+        if (startsWith != null) {
+            for (int i = 0; i < startsWith.length; i++) {
+                byte sB = startsWith[i];
+                if (token != null && sB != token[i]) {
+                    offsetInfo[3] = 0;
+                    break;
+                }
+            }
+        }
+        return offsetInfo;
+    }
+
+    /**
+     *
+     */
+    private int isEol(byte currentByte, int currentIndex) {
+        int crlfLength = 0;
+        if (currentByte == '\n') {
+            crlfLength = 1;
+        } else if (currentByte == '\r') {
+            if ((currentIndex + 1) >= this.bufferLength) {
+                this.index = currentIndex + 1;
+                this.fill();
+            }
+            currentByte = this.buffer[currentIndex + 1];
+            crlfLength = currentByte == '\n' ? 2 : 1;
+        }
+        return crlfLength;
+    }
+
+    /**
+     *
+     */
+    private byte[] extractDataToken(int length) {
+        byte[] data = null;
+        if (length > 0) {
+            data = new byte[length];
+            System.arraycopy(this.buffer, this.mark, data, 0, data.length);
+        }
+        return data;
+    }
+
+    /**
+     * Will fill the current buffer from current 'index' position, expanding it
+     * and or shuffling it if necessary
+     */
+    private void fill() {
+        if (this.index >= this.buffer.length) {
+            if (this.mark == 0) { // expand
+                byte[] newBuff = new byte[this.buffer.length + this.initialBufferSize];
+                System.arraycopy(this.buffer, 0, newBuff, 0, this.buffer.length);
+                this.buffer = newBuff;
+            } else { // shuffle
+                int length = this.index - this.mark;
+                System.arraycopy(this.buffer, this.mark, this.buffer, 0, length);
+                this.index = length;
+                this.mark = 0;
```

Comment:
I believe this.bufferLength should be set to length here, no? This is the same bug that was found recently in StreamDemarcator.